x-pack/plugins/security/server/authorization/roles/elasticsearch_role.ts

```diff
@@ -65,10 +68,16 @@ function transformRoleApplicationsToKibanaPrivileges(
   application: string,
   logger: Logger
 ) {
+  const isReservedPrivilege = (app: string): boolean => {
+    return (
+      app === RESERVED_PRIVILEGES_APPLICATION_WILDCARD || app === RESERVED_PRIVILEGES_ALL_WILDCARD
+    );
+  };
+
```

Comment:
I know this isn't ready for review, so I'm sorry for the premature review comment 🙂.

**nit**: "Reserved Privileges" is a legacy concept that is unrelated to the "All Wildcard". I recommend renaming `RESERVED_PRIVILEGES_ALL_WILDCARD` to something else, and decoupling the two. `PRIVILEGES_ALL_WILDCARD` might be good enough.

Reply:
Fixed and cherry-picked to a separate PR https://github.com/elastic/kibana/pull/181400